BlueprintUI/Sources/Layout/Layout.swift

```diff
@@ -33,10 +129,172 @@ public protocol Layout {
 
 }
 
-extension Layout where Traits == () {
+extension LegacyLayout where Traits == () {
 
     public static var defaultTraits: () {
         return ()
     }
 
 }
+
+public protocol CaffeinatedLayout {
+
+    typealias Subelements = LayoutSubelements
+
+    /// Cached values associated with the layout instance.
+    ///
+    /// If you create a cache for your custom layout, you can use a type alias to define this type
+    /// as your data storage type. Alternatively, you can refer to the data storage type directly in
+    /// all the places where you work with the cache.
+    ///
+    /// See ``makeCache(subelements:)-4kmy2`` for more information.
+    associatedtype Cache = Void
+
+    /// Returns the size of the composite element, given a proposed size constraint and the
+    /// container's subelements.
+    ///
+    /// Implement this method to tell your custom layout container’s parent how much space the
+    /// container needs for a set of subelements, given a size constraint. The parent might call
+    /// this method more than once during a layout pass with different proposed sizes to test the
+    /// flexibility of the container.
+    ///
+    /// In Blueprint, parents ultimately choose the size of their children, so the actual size that
+    /// this container is laid out in may not be a size that was returned from this method.
+    ///
+    /// ## Sizing Rules
+    ///
+    /// For performance reasons, the layout engine may deduce the measurement of your container for
+    /// some constraint values without explicitly calling
+    /// ``sizeThatFits(proposal:subelements:cache:)``. To ensure that the deduced value is correct,
+    /// your layout must follow some ground rules:
```

Comment:
Just a thought: I wonder if it's possible to implement a general fuzzer that can be run against layout implementations to ensure they adhere to the rules.

Reply:
Yes! We've got this on the list of potential follow-up work. It would be rather slow, so it would have to be an opt-in thing or a special test harness.